### PR TITLE
OCPBUGS-15906: ccoctl azure delete to also delete role assignments

### PR DESCRIPTION
**Pull request by jstuever**
The ccoctl azure subcommand was previously leaking role assignments. This change ensures all associated roleAssignments are removed prior to deleting the user-assigned managed identities.

### CODE REVIEW

File: pkg/cmd/provisioning/azure/delete.go [hunk 1]
```diff
@@ -80,6 +81,14 @@ func deleteManagedIdentities(client *azureclients.AzureClientWrapper, name, reso
 	for listManagedIdentities.More() {
 		pageResponse, err := listManagedIdentities.NextPage(context.Background())
 		if err != nil {
+			var respErr *azcore.ResponseError
+			if !(errors.As(err, &respErr)) {
+				return err
+			}
+			if respErr.ErrorCode == "ResourceGroupNotFound" {
+				log.Printf("Found no user-assigned managed identities in resource group %s", resourceGroupName)
```
Comment (abutcher) [inline pkg/cmd/provisioning/azure/delete.go:89]: Prefer if this said something about the resource group already being gone / having been deleted. Something like this similar to the latter `"Found no resource group %s"` log.

```suggestion
				log.Printf("Found no resource group %s. No user-assigned managed identities to list", resourceGroupName)
```

**Review by abutcher — COMMENTED** [review]
Works in my testing! One comment about the info log for no identities if that could be more specific to the RG not existing. Looks good

File: pkg/cmd/provisioning/azure/delete.go (hunk 1 above)
Comment (jstuever) [reply pkg/cmd/provisioning/azure/delete.go]: [Done](https://github.com/openshift/cloud-credential-operator/compare/cc1d63226f953c4ada6539888746cf4cbe463e1f..f83c6b70da024ca5e159d2515ab149227bad58d9)

**Pull request merged**